mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -86,6 +86,22 @@ private[ml] class WeightedLeastSquares(
     val aaBar = summary.aaBar
     val aaValues = aaBar.values
 
+    if (bStd == 0) {
+      if (fitIntercept) {
+        logWarning(s"The standard deviation of the label is zero, so the coefficients will be " +
+          s"zeros and the intercept will be the mean of the label; as a result, " +
+          s"training is not needed.")
+        val coefficients = new DenseVector(Array.ofDim(k-1))
+        val intercept = bBar
+        val diagInvAtWA = new DenseVector(Array.ofDim(k))
```

Comment:
Looking at the `LinearRegression` class, it handles the case where `bStd` == 0 by returning `DenseVector(Array(0D))` for `diagInvAtWA`. Then, in the training summary there are several cases to catch this invalid value for `diagInvAtWA`. This should be changed to:

```scala
val diagInvAtWA = new DenseVector(Array(0D))
```